### PR TITLE
[GPU] Pass activation unit tests on DG2

### PR DESCRIPTION
**Pull request by tuxedcat**
### Details:
 - Change Scale to Eltwise( mode::prod ) in unit test topologies.
 - Support Activation-Eltwise fusing.

### Tickets:
 - 67491

### CODE REVIEW

**Comment by sungeunk** [comment]
looks good to me

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/activation/activation_kernel_base.cpp [hunk 1]
```diff
@@ -28,7 +28,7 @@ ActivationKernelBase::DispatchData ActivationKernelBase::SetDefault(const activa
         dispatchData.gws = {out.X().v * out.Y().v, Align(out.Feature().v, 16), Align(out.Batch().v, 16)};
         dispatchData.lws = {1, 16, 16};
     } else {
-        dispatchData.gws = {out.X().v, out.Y().v * out.Z().v, out.Feature().v * out.Batch().v};
+        dispatchData.gws = {out.X().v*out.W().v, out.Y().v * out.Z().v, out.Feature().v * out.Batch().v};
         std::vector<std::vector<Tensor::DataChannelName>> dims_by_gws = {{Tensor::DataChannelName::X},
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/activation/activation_kernel_base.cpp:32]: isn't it necessary to add `Tensor::DataChannelName::W` too here?

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/activation/activation_kernel_base.cpp [hunk 2]
```diff
@@ -28,7 +28,7 @@ ActivationKernelBase::DispatchData ActivationKernelBase::SetDefault(const activa
         dispatchData.gws = {out.X().v * out.Y().v, Align(out.Feature().v, 16), Align(out.Batch().v, 16)};
         dispatchData.lws = {1, 16, 16};
     } else {
-        dispatchData.gws = {out.X().v, out.Y().v * out.Z().v, out.Feature().v * out.Batch().v};
+        dispatchData.gws = {out.X().v*out.W().v, out.Y().v * out.Z().v, out.Feature().v * out.Batch().v};
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/activation/activation_kernel_base.cpp:31]: please add a blank before and after the asterisk. (why isn't it caught during cpplint?)

File: src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/activation_ref.cl [hunk 3]
```diff
@@ -23,13 +23,27 @@ KERNEL(activation)(
 #endif
     )
 {
-#if OUTPUT_DIMS == 5
+#if OUTPUT_DIMS==6
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/activation_ref.cl:26]: Please add space before and after `==`

File: src/plugins/intel_gpu/tests/fusions/activation_fusion_test.cpp [hunk 4]
```diff
@@ -291,7 +291,28 @@ INSTANTIATE_TEST_SUITE_P(fusings_gpu, activation_scale_activation, ::testing::Va
...
+    tolerance = 1e-05f;
+    execute(p);
+}
+
+#define CASE_ACTIVATION_4D_F32_0 { 1, 2, 3, 2, 3, 2 }, data_types::f32, format::bfwzyx, data_types::f32, format::bfwzyx
+INSTANTIATE_TEST_SUITE_P(fusings_gpu, activation_eltwise, ::testing::ValuesIn(std::vector<activation_test_params>{
+    // InputDataType = FP32
+    activation_test_params{ CASE_ACTIVATION_4D_F32_0, 2, 3, "activation_ref" },
+    activation_test_params{ CASE_ACTIVATION_4D_F32_0, 2, 3, "activation_opt" }
+}));
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/activation_fusion_test.cpp:318]: Please add an empty line at the end of file.

File: src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp [hunk 5]
```diff
@@ -8,13 +8,98 @@
...
+    int x_size = output_layout.spatial(0);
+    int f_size = output_layout.feature();
+    int b_size = output_layout.batch();
+    auto bfwzyx = std::vector<int>{b_size, f_size, w_size, z_size, y_size, x_size};
+    EXPECT_EQ(output_layout.format, format::bfwzyx);
+    EXPECT_EQ(bfwzyx, shape);
+    for (size_t i = 0; i < input_raw.size(); ++i)
+        EXPECT_FLOAT_EQ(abs(input_raw[i])*elt_dat_raw[i], output_ptr[i]);
+}
+
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp:102]: can we reduce the code duplication between activation_ref test and activation_opt test?

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/activation/activation_kernel_base.cpp (hunk 1 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/activation/activation_kernel_base.cpp]: You're right. Fixed.

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/activation/activation_kernel_base.cpp (hunk 2 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/activation/activation_kernel_base.cpp]: Fixed.

File: src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/activation_ref.cl (hunk 3 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/activation_ref.cl]: Fixed.

File: src/plugins/intel_gpu/tests/fusions/activation_fusion_test.cpp (hunk 4 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/activation_fusion_test.cpp]: Fixed.

File: src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp (hunk 5 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp:62]: Removed duplicated codes as a function.

**Review by isanghao — APPROVED** [review]

**Pull request merged**